### PR TITLE
Multi asset 2

### PR DESCRIPTION
**Pull request by Robiquet**
- Refactor usd price fetching
- Fetch foreign asset metadata
- Support trading of foreign asset based market
- Support Liquidity management
- Support buy/sell full sets
- Fix Market page
- Fix Pool page
- Basic fixes for Portfolio page
- Fix market cards

### CODE REVIEW

**Review by TvrtkoM — CHANGES_REQUESTED** [review]
It seems that everything works except the market card not hiding skeleton for participants and liquidity for markets without a pool

File: components/markets/market-card/index.tsx [hunk 1]
```diff
@@ -140,7 +142,7 @@ const MarketCardDetails = ({
         </span>
       </div>
       <div className="flex gap-2.5 text-sm min-w-full">
-        {rows.numParticipants != null ? (
+        {rows.numParticipants != null && rows.baseAsset ? (
```
[the remark this answers is not in the document]
Comment (Robiquet) [reply components/markets/market-card/index.tsx:145]: Looks like we are fetching base asset from the pool, I'll change it to fetch from the market instead

File: lib/hooks/queries/useMarket.ts [hunk 2]
```diff
@@ -15,7 +15,11 @@ export const useMarket = (filter?: UseMarketFilter) => {
   const query = useQuery(
     [id, marketsRootQuery, filter],
     async () => {
-      if (isIndexedSdk(sdk)) {
+      if (
+        isIndexedSdk(sdk) &&
+        (("marketId" in filter && filter.marketId != null) ||
```
[the remark this answers is not in the document]
Comment (Robiquet) [reply lib/hooks/queries/useMarket.ts:20]: Looks like this gives a type error 
image

[the remark this answers is not in the document]
Comment (TvrtkoM) [reply lib/hooks/queries/useMarket.ts:20]: > Looks like this gives a type error

Oh :) didn't think that would happen.

**Review by TvrtkoM — APPROVED** [review]

**Pull request merged**